### PR TITLE
Sshrayber group nft

### PR DESCRIPTION
**Pull request by HagiaSofiya**

### CODE REVIEW

**Comment by HagiaSofiya** [comment]
Screenshot 2022-12-11 at 11 09 51 PM

File: components/profile/PublicProfile.tsx [hunk 1]
```diff
@@ -211,6 +215,27 @@ export default function PublicProfile(props: UserDetailsProps) {
     }
   }, [collabError]);
 
+  function groupBy<T>(arr: T[], fn: (item: T) => any) {
+    return arr.reduce<Record<string, T[]>>((prev, curr) => {
+      const groupKey = fn(curr);
+      const group = prev[groupKey] || [];
+      group.push(curr);
+      return { ...prev, [groupKey]: group };
```
Comment (msadura) [inline components/profile/PublicProfile.tsx:223]: please always avoid using spreading obejcts in reduce https://prateeksurana.me/blog/why-using-object-spread-with-reduce-bad-idea/

```
acc[groupKey] = group;
return acc;
```

should do the job here

**Review by msadura — CHANGES_REQUESTED** [review]

**Comment by msadura** [comment]
UI-wise I would make less verical sapcing between groups

**Review by mattcasey — APPROVED** [review]
Approved but agree with Marek's request - can you remove the spread inside reduce? Thanks!

File: components/profile/PublicProfile.tsx (hunk 1 above)
Comment (HagiaSofiya) [reply components/profile/PublicProfile.tsx:223]: @msadura Thank you for the article! I am not following the change you are asking for.  Would you mind sending me a sample?

Comment (msadura) [reply components/profile/PublicProfile.tsx:223]: sure:

```
  function groupBy<T>(arr: T[], fn: (item: T) => any) {
    return arr.reduce<Record<string, T[]>>((acc, curr) => {
      const groupKey = fn(curr);
      const group = prev[groupKey] || [];
      group.push(curr);
      
      acc[groupKey] = group;
      return acc;
    }, {});
 ```

**Review by msadura — APPROVED** [review]

**Comment by mattcasey** [comment]
we decided to nix this design for now, and the dev working on it has disappeared

**Pull request closed without merging**